scopes/ui-foundation/tree/folder-tree-node/folder-tree-node.tsx

```diff
@@ -1,18 +1,26 @@
 import { Icon } from '@teambit/design.elements.icon';
 import classNames from 'classnames';
 import React, { ReactNode, useState, useEffect } from 'react';
-import AnimateHeight from 'react-animate-height';
 
-import { TreeLayer, TreeNodeProps } from '@teambit/base-ui.graph.tree.recursive-tree';
-import { indentClass, indentStyle } from '@teambit/base-ui.graph.tree.indent';
+import { TreeNodeProps } from '@teambit/base-ui.graph.tree.recursive-tree';
+import { indentClass } from '@teambit/base-ui.graph.tree.indent';
+import { CollapsableTreeNode } from '@teambit/base-ui.graph.tree.collapsable-tree-node';
+import type { FolderPayload } from '@teambit/base-ui.graph.tree.collapsable-tree-node';
 import styles from './folder-tree-node.module.scss';
 
-export type FolderTreeNodeProps = {
-} & TreeNodeProps<FolderPayload>;
+export type FolderTreeNodeProps = {} & TreeNodeProps<FolderPayload>;
 
-export type FolderPayload = {
-  icon?: string | ReactNode;
-  open?: boolean;
+function getCustomIcon(icon: string | ReactNode) {
```

Comment:
it's ok, but hoisting functions to the top of the file is an automatic feature of JS.
I'd rather have the main focus of this file come fist, instead of utils.

I think we can disable this rule.